File: mini_gplus/models/link_preview.py
from enum import Enum
from mongoengine import Document, StringField, EnumField, URLField, ListField


class LinkPreviewState(Enum):
    Fetching = "fetching"
    Fetched = "fetched"
    Errored = "errored"


class LinkPreview(Document):
    url = URLField(required=True, unique=True)
    title = StringField(required=False, default='')
    subtitle = StringField(required=False, default='')
    image_urls = ListField(URLField, required=False, default=[])
    state = EnumField(LinkPreviewState, required=True)

File: web/src/models/LinkPreview.ts
type LinkPreviewState = 'fetching' | 'fetched' | 'errored'

export default interface LinkPreview {
  url: string
  title: string
  subtitle: string
  image_urls: string[]
  state: LinkPreviewState
  errored_next_refetch_seconds: number,
  index_start: number
  index_end: number
}
